client/components/common/tce-core/AddElement/index.vue

```diff
@@ -137,33 +167,41 @@ export default {
     },
     processedWidth() {
       return this.elementWidth === 50 ? LAYOUT.HALF_WIDTH : LAYOUT.FULL_WIDTH;
+    },
+    allowedTypes() {
+      const { elementWidth, include, layout, registry } = this;
+      if (!layout) return include;
+      const allowedElements = elementWidth === DEFAULT_ELEMENT_WIDTH
+        ? registry
+        : reject(registry, 'ui.forceFullWidth');
+      const allowedTypes = allowedElements.map(it => it.type);
+      return include ? intersection(include, allowedTypes) : allowedTypes;
     }
   },
   methods: {
-    add({ type, subtype, initState = () => ({}) }) {
-      const element = { type, data: { width: this.processedWidth } };
-      // If content element within activity
-      if (this.activity) {
-        element.activityId = this.activity.id;
-        element.position = this.position;
-      } else {
-        // If embed, assign id
-        element.id = uuid();
-        element.embedded = true;
-      }
-      if (isQuestion(element.type)) {
-        const data = { width: LAYOUT.FULL_WIDTH };
-        const question = [{ id: uuid(), data, type: 'JODIT_HTML', embedded: true }];
-        element.data = { ...element.data, question, type: subtype };
-      }
-      element.data = { ...element.data, ...initState() };
-      if (element.type === 'REFLECTION') delete element.data.correct;
-      this.$emit('add', element);
+    addElements(elements) {
+      const items = elements.map((it, index) => {
+        const position = this.position + index;
```

Comment:
if inserting multiple items between elements it needs to be bellow max as well ;)

Reply:
Reworked `calculatePosition` util with `AddElement`'s component logic, there were certain problems with inserting multiple elements I wasn't aware of 👀 .